src/libstd/lib.rs

```diff
@@ -471,6 +471,13 @@ pub mod time;
 mod sys_common;
 mod sys;
 
+#[unstable(feature = "demangle_for_rustc", issue = "0")]
+#[doc(hidden)]
+pub mod demangle_for_rustc {
+    pub use sys_common::backtrace::demangle;
+    pub use sys_common::backtrace::PrintFormat;
+}
```

Comment:
Well there’s technically a [rustc-demangle](https://crates.io/crates/rustc-demangle) already, although I do not know how easy/hard it is to integrate an external crate into libstd. I know there are a few already integrated in other places such as librustc_trans, could take a look there.